runtime/ast/expression_extractor.go

```diff
@@ -781,3 +786,35 @@ func (extractor *ExpressionExtractor) VisitPathExpression(expression *PathExpres
 func (extractor *ExpressionExtractor) ExtractPath(expression *PathExpression) ExpressionExtraction {
 	return rewriteExpressionAsIs(expression)
 }
+
+func (extractor *ExpressionExtractor) VisitAttachExpression(expression *AttachExpression) ExpressionExtraction {
+
+	// delegate to child extractor, if any,
+	// or call default implementation
+
+	if extractor.AttachExtractor != nil {
+		return extractor.AttachExtractor.ExtractAttach(extractor, expression)
+	}
+	return extractor.ExtractAttach(expression)
+}
+
+func (extractor *ExpressionExtractor) ExtractAttach(expression *AttachExpression) ExpressionExtraction {
+	// copy the expression
+	newExpression := *expression
```

Comment:
I think it would be clearer if this happened after setting `rewrittenExpressions`. I'd also find it clearer if `newExpression` were created whole-cloth instead of copying it then rewriting 2/3 of its fields.

Reply:
This is the style the rest of the functions in this file use, so I just copied from that. cc @turbolent is there a reason we do things this way?